### PR TITLE
[mlir] Remove "Syntax:" parser where it's already provided by `assemblyFormat`

### PR DESCRIPTION
**Pull request by alexbeloi**
See #73359

Types using `assemblyFormat` to define parsing don't need an additional
handwritten parser. So we should remove the handwritten parsers where one
provided by an `assemblyFormat` already exists to avoid confusion and
de-syncing.

### CODE REVIEW

**Comment by alexbeloi** [comment]
@kuhar

File: mlir/include/mlir/Dialect/UB/IR/UBOps.td [hunk 1]
```diff
@@ -48,15 +48,6 @@ def PoisonOp : UB_Op<"poison", [ConstantLike, Pure]> {
...
 
-    Syntax:
-
-    ```
-    poison-op ::= `poison` (`<` value `>`)? `:` type
-    ```
-
-    Examples:
-
-    ```
```
Comment (kuhar) [inline mlir/include/mlir/Dialect/UB/IR/UBOps.td:59]: This should not have been removed

**Review by kuhar — COMMENTED** [review]
Looks great, just one comment

File: mlir/include/mlir/Dialect/UB/IR/UBOps.td (hunk 1 above)
Comment (alexbeloi) [reply mlir/include/mlir/Dialect/UB/IR/UBOps.td]: Thanks for catching, reverted.

**Review by kuhar — APPROVED** [review]
LGTM

**Comment by alexbeloi** [comment]
I don't have write access to the repo, so how does this get merged?

**Comment by kuhar** [comment]
> I don't have write access to the repo, so how does this get merged?

We should give it some time in case someone else wants review. If there are changes requested in the next 24h I will merge it for you. The overall process is described here: https://llvm.org/docs/CodeReview.html and https://llvm.org/docs/DeveloperPolicy.html#obtaining-commit-access.

**Pull request merged**